pic-sure-resources/pic-sure-aggregate-data-sharing-resource/src/main/java/edu/harvard/hms/dbmi/avillach/AggregateDataSharingResourceRS.java

```diff
@@ -138,73 +138,64 @@ public ResourceInfo info(QueryRequest infoRequest) {
 	@Override
 	public SearchResults search(QueryRequest searchRequest) {
 		logger.debug("Calling Aggregate Data Sharing Search");
-		if (searchRequest == null || searchRequest.getQuery() == null) {
-			throw new ProtocolException(ProtocolException.MISSING_DATA);
-		}
-
-		String pathName = "/search";
-		try {
-			QueryRequest chainRequest = new QueryRequest();
-			chainRequest.setQuery(searchRequest.getQuery());
-			chainRequest.setResourceCredentials(searchRequest.getResourceCredentials());
-
-			if(properties.getTargetResourceId() != null && !properties.getTargetResourceId().isEmpty()) {
-				chainRequest.setResourceUUID(UUID.fromString(properties.getTargetResourceId()));
-			} else {
-				chainRequest.setResourceUUID(searchRequest.getResourceUUID());
-			}
-
-			String payload = objectMapper.writeValueAsString(chainRequest);
-			String composedURL = HttpClientUtil.composeURL(properties.getTargetPicsureUrl(), pathName);
-			HttpResponse response = HttpClientUtil.retrievePostResponse(composedURL, headers, payload);
-			if (HttpClientUtil.is2xx(response)) {
-				logger.error("{}{} did not return a 200: {} {} ", properties.getTargetPicsureUrl(), pathName,
-						response.getStatusLine().getStatusCode(), response.getStatusLine().getReasonPhrase());
-				HttpClientUtil.throwResponseError(response, properties.getTargetPicsureUrl());
-			}
-			return readObjectFromResponse(response, SearchResults.class);
-		} catch (IOException e) {
-			// Note: this shouldn't ever happen
-			logger.error("Error encoding search payload", e);
-			throw new ApplicationException(
-					"Error encoding search for resource with id " + searchRequest.getResourceUUID());
-		}
+		checkQuery(searchRequest);
```

Comment:
Not thrilled about all the exception based control flow, but we'll leave that for another PR.